### PR TITLE
Fix some plural and puzzle strings, and remove other unused ones

### LINKED ISSUE #2085: Remove the puzzle strings
Reported by @SafaAlfulaij https://github.com/vector-im/riot-android/pull/1934#issuecomment-373441599

### PR DESCRIPTION
**Pull request by SafaAlfulaij**
This will fix some plural messages to be plural aware, puzzle strings to not be puzzle, and remove unused strings.

Tests: Not tested, based on logic.

Fixes #2085

Signed-off-by: Safa AlFulaij safa1996alfulaij@gmail.com

### CODE REVIEW

**Comment by SafaAlfulaij** [comment]
Alright, the errors now are that some Format strings are not valid, and I don't really know what this means, sorry.
Others are about strings being different in the translated values files (whether the string is new or changed). This is supposed to be updated correctly by Weblate, right?

File: vector/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -410,10 +413,18 @@
     <string name="room_participants_action_devices_list">Show Device List</string>
     <string name="room_participants_power_level_prompt">You will not be able to undo this change as you are promoting the user to have the same power level as yourself.\nAre you sure?</string>
 
-    <string name="room_participants_invite_prompt_msg">"Are you sure you want to invite %s to this chat?"</string>
     <string name="room_participants_kick_prompt_msg">Are you sure that you want to kick this user from this chat?</string>
     <string name="room_participants_ban_prompt_msg">Are you sure that you want to ban this user from this chat?</string>
 
+    <!-- The variable can be a single name or a string of names ("A, ""B, ""C and D" > "A, B, C and D) -->
```
Comment (bmarty) [inline vector/src/main/res/values/strings.xml]: I do not understand the usage of `"` in this comment. Is there some typo?
Maybe write an example for clarity: `"Bob" or "Bob and Alice" or "Bob, Alice and Charlie"`

File: vector/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -410,10 +413,18 @@
...
+    <string name="room_participants_invite_prompt_msg">"Are you sure you want to invite %s to this chat?"</string>
+    <!-- The building block of the first part of the string of names ("A, "/"B, ") -->
```
Comment (bmarty) [inline vector/src/main/res/values/strings.xml]: Same remark for this one

File: vector/src/main/res/values/strings.xml (hunk 1 above)
Comment (SafaAlfulaij) [reply vector/src/main/res/values/strings.xml]: Just to give the translator an idea how the block is built, to show that there is a space after the comma.
And yes, human names are better :)
Will work now on it.

**Comment by bmarty** [comment]
Also "Tests: Not tested, based on logic." is a bit scary no? ;)

**Comment by SafaAlfulaij** [comment]
“BUILD SUCCESSFUL in 3m 28s” :sob: 

Yes, “Tests: Not tested, based on logic.” is scary, but that is what you get when a translator works as a developer of a language he doesn't even fully understand :p
Anyway the auto test confirms that things are fine now :)

About the time strings change, maybe it's a bit too much, but I think it's better to give full control to the translator.
I'm pretty sure builtins in Android have something ready for this, not sure where…

**Comment by SafaAlfulaij** [comment]
@bmarty PING! :p
Please inform me about any other concerns.

File: vector/src/main/res/values-ar/strings.xml [hunk 3]
```diff
@@ -71,14 +71,6 @@
     <string name="rooms_directory_header">دليل الغرف</string>
     <string name="no_room_placeholder">لا غرف</string>
     <string name="no_public_room_placeholder">لا غرف عمومية متوفرة</string>
-    <plurals name="public_room_nb_users">
```
Comment (bmarty) [inline vector/src/main/res/values-ar/strings.xml]: I think this resource has been deleted by mistake, no? It's not deleted in the main `strings.xml` file. This remark is applicable to all other translations as well

**Comment by SafaAlfulaij** [comment]
I've returned them but changed the tabs into spaces (I thought it's my browser/GitHub doing this). Not sure why weblate will do that for these strings only as other strings are using normal spaces…

**Comment by bmarty** [comment]
Ok, thanks, I've merged the PR

**Comment by bmarty** [comment]
What do you mean? Can you figured out a resource Id in arabic that is problematic?
Thanks

**Pull request merged**